editor/editor_main_screen.cpp

```diff
@@ -0,0 +1,298 @@
+				}
+			}
+		} break;
+	}
+}
+
+void EditorMainScreen::set_button_container(HBoxContainer *p_button_hb) {
+	button_hb = p_button_hb;
+}
+
+void EditorMainScreen::save_layout_to_config(Ref<ConfigFile> p_config_file, const String &p_section) const {
+	int selected_main_editor_idx = -1;
+	for (int i = 0; i < buttons.size(); i++) {
+		if (buttons[i]->is_pressed()) {
+			selected_main_editor_idx = i;
+			break;
+		}
+	}
+	if (selected_main_editor_idx != -1) {
+		p_config_file->set_value(p_section, "selected_main_editor_idx", selected_main_editor_idx);
+	} else {
+		p_config_file->set_value(p_section, "selected_main_editor_idx", Variant());
+	}
+}
+
+void EditorMainScreen::load_layout_from_config(Ref<ConfigFile> p_config_file, const String &p_section) {
+	if (p_config_file->has_section_key(p_section, "selected_main_editor_idx")) {
+		int selected_main_editor_idx = p_config_file->get_value(p_section, "selected_main_editor_idx");
+		if (selected_main_editor_idx >= 0 && selected_main_editor_idx < buttons.size()) {
+			callable_mp(this, &EditorMainScreen::select).call_deferred(selected_main_editor_idx);
+		}
+	}
+}
+
+void EditorMainScreen::set_button_enabled(int p_index, bool p_enabled) {
+	ERR_FAIL_INDEX(p_index, buttons.size());
+	buttons[p_index]->set_visible(p_enabled);
+	if (!p_enabled && buttons[p_index]->is_pressed()) {
+		select(EDITOR_2D);
+	}
+}
+
+bool EditorMainScreen::is_button_enabled(int p_index) const {
+	ERR_FAIL_INDEX_V(p_index, buttons.size(), false);
+	return buttons[p_index]->is_visible();
+}
+
+int EditorMainScreen::_get_current_main_editor() const {
+	for (int i = 0; i < editor_table.size(); i++) {
+		if (editor_table[i] == selected_plugin) {
+			return i;
+		}
+	}
+
+	return 0;
+}
+
+void EditorMainScreen::select_next() {
+	int editor = _get_current_main_editor();
+
+	do {
+		if (editor == editor_table.size() - 1) {
+			editor = 0;
+		} else {
+			editor++;
+		}
+	} while (!buttons[editor]->is_visible());
+
+	select(editor);
+}
+
+void EditorMainScreen::select_prev() {
+	int editor = _get_current_main_editor();
+
+	do {
+		if (editor == 0) {
+			editor = editor_table.size() - 1;
+		} else {
+			editor--;
+		}
+	} while (!buttons[editor]->is_visible());
+
+	select(editor);
+}
+
+void EditorMainScreen::select_by_name(const String &p_name) {
+	ERR_FAIL_COND(p_name.is_empty());
+
+	for (int i = 0; i < buttons.size(); i++) {
+		if (buttons[i]->get_text() == p_name) {
+			select(i);
+			return;
+		}
+	}
+
+	ERR_FAIL_MSG("The editor name '" + p_name + "' was not found.");
+}
+
+void EditorMainScreen::select(int p_index) {
+	static bool selecting = false;
+	if (selecting || EditorNode::get_singleton()->is_changing_scene()) {
+		return;
+	}
+
+	ERR_FAIL_INDEX(p_index, editor_table.size());
+
+	if (!buttons[p_index]->is_visible()) { // Button hidden, no editor.
+		return;
+	}
+
+	selecting = true;
+
+	for (int i = 0; i < buttons.size(); i++) {
+		buttons[i]->set_pressed(i == p_index);
```

Comment:
```suggestion
		buttons[i]->set_pressed_no_signal(i == p_index);
```
Allows to remove `selecting`.